File: Core/Models/Shipment.cs
namespace Core.Models
{
    public class Shipment
    {
        public Guid Id { get; set; }
        public string Number { get; set; }
        public Guid ClientId { get; set; }
        public Client Client { get; set; }
        public DateTime Date { get; set; }
        public bool IsSigned {  get; set; } = false;

        public ICollection<ShipmentResource> ShipmentResources { get; set; }
    }
}

File: web/src/Entities/Shipment.ts
import { UUID } from "crypto";

interface Shipment {
    id: UUID;
    number: string;
    clientId: UUID;
    clientName: UUID;
    date: Date;
    isSigned: boolean
    resources: ShipmentResource[]
}

interface ShipmentResource {
    id: UUID;
    shipmentId: UUID;
    resourceId: UUID;
    resourceName: string;
    unitId: UUID;
    unitName: string;
    quantity: number
}

export type {Shipment, ShipmentResource};